File: Proyecto_edd/Producto.java
public class Producto {
    private int idProducto;
    private String nombre;
    private String descripcion;
    private float precio;
    private int stock;
    private String rutaFoto;

    // Constructor
    public Producto(int idProducto, String nombre, String descripcion, float precio, int stock, String rutaFoto) {
        this.idProducto = idProducto;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio >= 0 ? precio : 0;
        this.stock = stock >= 0 ? stock : 0;
        this.rutaFoto = rutaFoto;
    }

    // Getters
    public int getIdProducto() {
        return idProducto;
    }

    public String getNombre() {
        return nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public float getPrecio() {
        return precio;
    }

    public int getStock() {
        return stock;
    }

    public String getRutaFoto() {
        return rutaFoto;
    }

    // Setters
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public void setPrecio(float precio) {
        if (precio < 0) {
            throw new IllegalArgumentException("El precio no puede ser negativo.");
        }
        this.precio = precio;
    }

    public void setStock(int stock) {
        if (stock < 0) {
            throw new IllegalArgumentException("El stock no puede ser negativo.");
        }
        this.stock = stock;
    }

    public void setRutaFoto(String rutaFoto) {
        this.rutaFoto = rutaFoto;
    }

    // Método para actualizar el stock del producto
    public boolean actualizarStock(int cantidad) {
        if (stock + cantidad < 0) {
            System.out.println("No hay suficiente stock disponible para realizar esta operación.");
            return false;
        }
        stock += cantidad;
        return true;
    }

    // Representación en cadena del producto
    @Override
    public String toString() {
        return "Producto{" +
               "idProducto=" + idProducto +
               ", nombre='" + nombre + '\'' +
               ", descripcion='" + descripcion + '\'' +
               ", precio=" + precio +
               ", stock=" + stock +
               ", rutaFoto='" + rutaFoto + '\'' +
               '}';
    }
}

File: scytec/src/Producto.py
class Producto:
    def __init__(self, id_producto: int, nombre: str, descripcion: str, precio: float, stock: int, ruta_foto: str, id_tienda: int):
        """
        Constructor para inicializar un producto.
        :param id_producto: Identificador único del producto.
        :param nombre: Nombre del producto.
        :param descripcion: Descripción del producto.
        :param precio: Precio del producto.
        :param stock: Cantidad disponible del producto.
        :param ruta_foto: Ruta de la foto del producto.
        :param id_tienda: Identificador de la tienda asociada.
        """
        self.id_producto = id_producto
        self.nombre = nombre
        self.descripcion = descripcion
        self.precio = precio
        self.stock = stock
        self.ruta_foto = ruta_foto
        self.id_tienda = id_tienda

    def __str__(self):
        """
        Representación en cadena del producto.
        """
        return (f"Producto(ID: {self.id_producto}, Nombre: {self.nombre}, "
                f"Descripción: {self.descripcion}, Precio: ${self.precio:.2f}, "
                f"Stock: {self.stock}, Foto: {self.ruta_foto}, Tienda: {self.id_tienda})")

    def actualizar_stock(self, nuevo_stock: int):
        """
        Actualiza el stock del producto.
        :param nuevo_stock: Nueva cantidad de stock.
        """
        self.stock = nuevo_stock
        print(f"El stock del producto '{self.nombre}' se ha actualizado a {nuevo_stock}.")

    def aplicar_descuento(self, porcentaje: float):
        """
        Aplica un descuento al precio del producto.
        :param porcentaje: Porcentaje de descuento a aplicar.
        """
        descuento = self.precio * (porcentaje / 100)
        self.precio -= descuento
        print(f"Se aplicó un descuento del {porcentaje}%. Nuevo precio: ${self.precio:.2f}")
